### PR TITLE
RSDK-2313 add slam service name to motionRequest proto message

### PR DESCRIPTION
**Pull request by raybjork**
This pull request implements updates RDK to match the proto message defined in [this pull request](https://github.com/viamrobotics/api/pull/216)

### CODE REVIEW

File: testutils/inject/motion_service.go [hunk 1]
```diff
@@ -13,6 +13,14 @@ import (
...
 	motion.Service
 	MoveFunc func(
+		ctx context.Context,
+		componentName resource.Name,
+		grabPose *referenceframe.PoseInFrame,
+		worldState *referenceframe.WorldState,
+		slamName resource.Name,
+		extra map[string]interface{},
+	) (bool, error)
+	MoveSingleComponentFunc func(
```
Comment (spieswl) [inline testutils/inject/motion_service.go:23]: Nice catch

**Review by spieswl — APPROVED** [review]
LGTM

File: go.mod [hunk 2]
```diff
@@ -409,3 +409,5 @@ require (
 	github.com/ziutek/mymysql v1.5.4 // indirect
 	golang.org/x/exp v0.0.0-20220722155223-a9213eeb770e // indirect
 )
+
+replace go.viam.com/api => github.com/raybjork/api v0.0.0-20230310213510-f207dfec088c
```
Comment (spieswl) [inline go.mod:413]: Dont forget to clean this one up.

**Review by spieswl — CHANGES_REQUESTED** [review]

File: go.mod (hunk 2 above)
Comment (raybjork) [reply go.mod:413]: Yes will remember for sure!

**Pull request merged**